WordPress/Classes/ViewRelated/Domains/Domain registration/RegisterDomainSuggestions/RegisterDomainCoordinator.swift

```diff
@@ -0,0 +1,204 @@
+        proxy.createPersistentDomainShoppingCart(siteID: siteID,
+                                                 domainSuggestion: domain.remoteSuggestion(),
+                                                 privacyProtectionEnabled: domain.supportsPrivacy ?? false,
+                                                 success: { _ in
+            onSuccess()
+        },
+                                                 failure: { _ in
+            onFailure()
+        })
+    }
+
+    func presentWebViewForNoSite(on viewController: UIViewController) {
+        guard let domain,
+              let url = URL(string: Constants.noSiteCheckoutWebAddress) else {
+            crashLogger.logMessage("Failed to present domain checkout webview for no-site.",
+                                   level: .error)
+            return
+        }
+
+        presentCheckoutWebview(on: viewController,
+                               domainSuggestion: domain,
+                               url: url,
+                               title: TextContent.checkoutTitle,
+                               shouldPush: true)
+    }
+
+    func presentWebViewForCurrentSite(on viewController: UIViewController) {
+        guard let domain,
+              let site,
+              let homeURL = site.homeURL,
+              let siteUrl = URL(string: homeURL as String), let host = siteUrl.host,
+              let url = URL(string: Constants.checkoutWebAddress + host) else {
+            crashLogger.logMessage("Failed to present domain checkout webview for current site.",
+                                   level: .error)
+            return
+        }
+
+        presentCheckoutWebview(on: viewController,
+                               domainSuggestion: domain,
+                               url: url,
+                               title: nil,
+                               shouldPush: false)
+    }
+
+    func handleNoSiteChoice(on viewController: UIViewController) {
+        createCart(
+            onSuccess: { [weak self] in
+                self?.presentWebViewForNoSite(on: viewController)
+            }) {
+                viewController.displayActionableNotice(title: TextContent.errorTitle, actionTitle: TextContent.errorDismiss)
+            }
+    }
+
+    func handleExistingSiteChoice(on viewController: UIViewController) {
+        print("handleExistingSiteChoice")
+    }
+
+    // MARK: Helpers
+
+    private func presentCheckoutWebview(on viewController: UIViewController,
+                                        domainSuggestion: FullyQuotedDomainSuggestion,
+                                        url: URL,
+                                        title: String?,
+                                        shouldPush: Bool) {
+
+        let webViewController = WebViewControllerFactory.controllerWithDefaultAccountAndSecureInteraction(
+            url: url,
+            source: "domains_register", // TODO: Update source
```

Comment:
Don't forget to follow up on this @hassaanelgarem 👍

Reply:
I created this issue to track this: #21996 👍